main_test.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"net/http/httptest"
+	"strconv"
+	"testing"
+	"time"
+
+	"k8s.io/kube-state-metrics/pkg/options"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+func BenchmarkKubeStateMetrics(t *testing.B) {
+	kubeClient := fake.NewSimpleClientset()
+
+	if err := injectFixtures(kubeClient); err != nil {
+		t.Errorf("error injecting pod add: %v", err)
+	}
+
+	opts := options.NewOptions()
+	collectors := options.DefaultCollectors
+	namespaces := options.DefaultNamespaces
+
+	registry := prometheus.NewRegistry()
+	registerCollectors(registry, kubeClient, collectors, namespaces, opts)
+	handler := promhttp.HandlerFor(registry, promhttp.HandlerOpts{ErrorLog: promLogger{}})
+
+	req := httptest.NewRequest("GET", "http://localhost:8080/metrics", nil)
+
+	// TODO: necessary?
+	time.Sleep(time.Second)
+
+	amountRequests := 10
+	for i := 0; i < amountRequests; i++ {
+		w := httptest.NewRecorder()
+		handler.ServeHTTP(w, req)
+	}
+}
+
+func injectFixtures(client *fake.Clientset) error {
+	creators := []func(*fake.Clientset, int) error{
+		configMap,
+		pod,
```

Comment:
Happy to add more fixtures here, but would prefer general feedback first to prevent lost work.